File: TikraqApp/src/app/model/Consulta.ts
export class Consulta {
  id: number;
  titulo: string;
  descripcion: string;
  usuariosUsuarioid: { id: number };

  constructor(id: number=0, titulo: string, descripcion: string,usuariosUsuarioid: { id: number }) {
    this.id = id;
    this.titulo = titulo;
    this.descripcion = descripcion;
    this.usuariosUsuarioid = usuariosUsuarioid;
  }
}

File: Tikraq/src/main/java/com/example/tikraq/entities/Consulta.java
package com.example.tikraq.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "consultas")
public class Consulta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "consultaId")
    private Integer id;

    private String titulo;
    private String descripcion;
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "usuarios_usuarioid")
    private Usuario usuariosUsuarioid;
}
